pkg/api/v1/atlasdatabaseuser_types.go

```diff
@@ -168,3 +168,48 @@ func (p AtlasDatabaseUser) ToAtlas(kubeClient client.Client) (*mongodbatlas.Data
 
 	return result, err
 }
+
+// ************************************ Builder methods *************************************************
+
+func NewDBUser(namespace, name, dbUserName, projectName string) *AtlasDatabaseUser {
+	return &AtlasDatabaseUser{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: AtlasDatabaseUserSpec{
+			Username:       dbUserName,
+			Project:        ResourceRef{Name: projectName},
+			PasswordSecret: &ResourceRef{},
+			Roles:          []RoleSpec{},
+		},
+	}
+}
+
+func (p *AtlasDatabaseUser) WithName(name string) *AtlasDatabaseUser {
+	p.Name = name
+	return p
+}
+func (p *AtlasDatabaseUser) WithDBUserName(name string) *AtlasDatabaseUser {
```

Comment:
[nit] There are some inconsistencies: in Cluster & Project it's `WithAtlasName`, but here we have `WithDBUserName`. Maybe we could change this to `WithAtlasName` as well?